### PR TITLE
update TKR,Capabilities, Features, FeatureGates to use v1 CRD

### PR DESCRIPTION
**Pull request by yharish991**
Signed-off-by: Harish Yayi <yharish991@gmail.com>

### What this PR does / why we need it

This PR upgrades TKR,Capabilities, Features, FeatureGates, StorageClasses to use v1 CRD

### Which issue(s) this PR fixes
Fixes https://github.com/vmware-tanzu/tanzu-framework/issues/1097

### Describe testing done for PR

1. make build-install-cli-local 
2. tanzu management-cluster create -v 6 --dry-run 
3. Inspect for capabilities CRS

### Release note

```release-note
Update TKR, Capabilities, Features, FeatureGates to use v1 CustomResourceDefintion and StorageClasses for Kubernetes v1.22 compatibility
```

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

File: pkg/v1/providers/ytt/03_customizations/capabilities/add_capabilities.yaml [hunk 1]
```diff
@@ -479,6 +458,10 @@ spec:
         - effect: NoSchedule
           key: node-role.kubernetes.io/master
 #@ end
+
+#@ if data.values.DISABLE_CRS_FOR_ADDON_TYPE and "capabilities/capabilities-controller" in data.values.DISABLE_CRS_FOR_ADDON_TYPE:
```
Comment (rajathagasthya) [inline pkg/v1/providers/ytt/03_customizations/capabilities/add_capabilities.yaml:462]: This looks unrelated. Is this a bug we are fixing?

**Review by rajathagasthya — COMMENTED** [review]
Quick question: are we updating Feature and FeatureGate CRDs in a separate PR?

File: pkg/v1/providers/ytt/03_customizations/capabilities/add_capabilities.yaml (hunk 1 above)
Comment (yharish991) [reply pkg/v1/providers/ytt/03_customizations/capabilities/add_capabilities.yaml:456]: Added this logic to handle the scenario where the existing management cluster is upgraded to the new version and capabilities CRD still is on v1beta1. In the upgraded cluster we need to upgrade the Capabilities CRD too else it will break

**Review by rajathagasthya — APPROVED** [review]

**Review by danniel1205 — APPROVED** [review]
LGTM

**Review by rajathagasthya — APPROVED** [review]

**Review by jmoroski — APPROVED** [review]

**Comment by ashish-amarnath** [comment]
@yharish991 can you please incorporate changes in https://github.com/vmware-tanzu/tanzu-framework/pull/1133 into this PR as well?

**Review by imikushin — APPROVED** [review]
